dash-renderer/src/observers/loadingMap.ts

```diff
@@ -43,26 +42,22 @@ const observer: IStoreObserverDefinition<IStoreState> = {
         const nextMap: any = isEmpty(loadingPaths) ?
             null :
             reduce(
-                (res, path) => {
+                (res, {id, property, path}) => {
                     let target = res;
-                    const idprop = {
-                        id: path.id,
-                        property: path.property
-                    };
+                    const idprop = {id, property};
 
                     // Assign all affected props for this path and nested paths
                     target.__dashprivate__idprops__ = target.__dashprivate__idprops__ || [];
                     target.__dashprivate__idprops__.push(idprop);
 
-                    forEach(p => {
-                        target = (target[p] =
-                            target[p] ??
-                                p === 'children' ? [] : {}
```

Comment:
Turns out `??` has higher precedence than `?` in JS 🙈 The only necessary part of this fix is the extra parentheses around this line.

Added the `typeof path[i + 1]` clause so that the `[]` vs `{}` distinction is correct - though it doesn't seem to actually cause problems when the container has the wrong type 🤷